### DISCUSSION TITLE
Multiple DDS Domains in one Subscriber Program

### DISCUSSION THREAD

**Discussion by mdavies2310** [OpenDDS/OpenDDS#4282] [Q&A] [answered]
Is there a way to have a subscriber program look on multiple domains? My program now has two participants, two subscribers. The topics will be the same on both domains. It seems like it only gets the data from the domain with more coming through. For example:

DDSDomain 0 -> A lot of data
DDSDomain 1 -> Not much data, slower rate

If I run this this I only get data from DDSDomain 0.

If I switch domains
DDSDomain 0 -> Not much data, slower rate
DDSDomain 1 -> A lot of data

I only get data from DDSDomain 1

If I comment out the domain with alot of data, I get the data from the other domain.

DDSDomain 0 -> Not much data, slower rate
DDSDomain 1 -> Not used

**Comment by mitza-oci** [comment]
Yes, multiple domain participants can coexist and be in different domains.  Configuring this depends on the transport type you've selected, see https://github.com/OpenDDS/OpenDDS/issues/270

**Comment by mdavies2310** [comment]
I have multiples of everything, and it is still not working. I do literally everything two times.
 Is there a specific setting I need to set to enable this? I am using DDS RTPS.

I read the mentioned link but did not see an answer in there.

**Reply by mitza-oci** [reply]
When using the rtps_udp transport, make sure there are two distinct transport instances with each one used by a different domain participant.

Based on your description I'm pretty sure that's the problem.  But if it's not please attach a log file.

**Reply by mdavies2310** [reply]
What do you mean by transport instances? I need to run two of the subscriber exes at the same time?

**Reply by mitza-oci** [reply, marked as answer]
No, transport instances are objects in the same process as your code.
https://opendds.readthedocs.io/en/latest-release/devguide/run_time_configuration.html#transport-concepts

**Reply by mdavies2310** [reply]
I figured it out. Thanks!
